components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -953,15 +949,24 @@ private static boolean isResourceEpUnavailable(Operation operation) {
         return false;
     }
 
-    public static List<String> setAuthProviders(BasicAuth basicAuth) {
+    public static List<String> getAuthProviders(String schemas) {
         List<String> authProviders = new ArrayList<>();
-        if (basicAuth != null && basicAuth.isOptional()) {
-            authProviders.add(APISecurity.basic.name());
-            authProviders.add(APISecurity.oauth2.name());
-            authProviders.add(APISecurity.jwt.name());
-        } else if (basicAuth != null && basicAuth.isRequired()) {
-            authProviders.add(APISecurity.basic.name());
-        } else {
+        boolean basic = false;
+        boolean oauth2 = false;
+        if (schemas != null) {
+            String[] schemasArray = schemas.trim().split("\\s*,\\s*");
+            for (String s : schemasArray) {
+                if (s.equalsIgnoreCase(APISecurity.basic.name())) {
+                    authProviders.add(APISecurity.basic.name());
+                } else if (s.equalsIgnoreCase(APISecurity.apikey.name())) {
+                    authProviders.add(APISecurity.apikey.name());
+                } else if (s.equalsIgnoreCase(APISecurity.oauth2.name())) {
+                    authProviders.add(APISecurity.oauth2.name());
+                    authProviders.add(APISecurity.jwt.name());
+                }
+            }
+        }
+        if (authProviders.size() < 1) {
```

Comment:
please use isEmpty()

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/973